crates/ruff/src/importer/insertion.rs

```diff
@@ -67,9 +67,13 @@ impl<'a> Insertion<'a> {
             TextSize::default()
         };
 
-        // Skip over commented lines.
+        // Skip over commented lines, with whitespace separation.
         for line in UniversalNewlineIterator::with_offset(locator.after(location), location) {
-            if line.trim_whitespace_start().starts_with('#') {
+            let trimmed_line = line.trim_whitespace_start();
+            if trimmed_line.is_empty() {
+                continue;
+            }
+            if trimmed_line.starts_with('#') {
```

Comment:
Given:

```python
# This is a comment

# This is a comment about this specific import
from sys import stdout
```

I guess we'd now do:

```python
# This is a comment

# This is a comment about this specific import
from __future__ import annotations
from sys import stdout
```

I guess we arguably already had that problem, since given:
```python
# This is a comment about this specific import
from sys import stdout
```

We already do:

```python
# This is a comment about this specific import
from __future__ import annotations
from sys import stdout
```

I suppose we could modify the logic to only skip comments if the comments are followed by a blank line but that may be surprising -- I'm not sure. What do you think?

Reply:
That's a good point. For what it's worth, `isort -a "from __future__ import annotations"` seems to do the same thing:

```
# This is a comment

# Comment about this import
from foo import bar
```

turns into:

```
# This is a comment

# Comment about this import
from __future__ import annotations

from foo import bar
```

That makes me inclined to think this case isn't too valuable to support. If we do want to support it, though, I do think your proposal of skipping comments if the comments are followed by a blank line is pretty reasonable and intuitive.

Reply:
Ok, thanks for looking into that. If `isort` has this behavior, then it seems reasonable to me to proceed as-is.